solidity_contracts/src/CairoPrecompiles/DualVmToken.sol

```diff
@@ -7,43 +7,62 @@ using CairoLib for uint256;
 
 /// @notice EVM adapter into a Cairo ERC20 token
 /// @dev This implementation is highly experimental
+///      It relies on CairoLib to perform Cairo precompile calls
+///      Events are emitted in this contract but also in the Starknet token contract
 /// @author Kakarot
 /// @author Modified from Solmate (https://github.com/transmissions11/solmate/blob/main/src/tokens/ERC20.sol)
-/// @dev Do not manually set balances without updating totalSupply, as the sum of all user balances must not exceed it.
 contract DualVmToken {
     /*//////////////////////////////////////////////////////////////
                         CAIRO SPECIFIC VARIABLES
     //////////////////////////////////////////////////////////////*/
 
+    /// @dev The prime number used in the Starknet field
+    uint256 public constant STARKNET_FIELD_PRIME = 2 ** 251 + 17 * 2 ** 192 + 1;
+
     /// @dev The address of the starknet token to call
-    uint256 immutable starknetToken;
+    uint256 public immutable starknetToken;
 
     /// @dev The address of the kakarot starknet contract to call
-    uint256 immutable kakarot;
+    uint256 public immutable kakarot;
 
     /*//////////////////////////////////////////////////////////////
                                  EVENTS
     //////////////////////////////////////////////////////////////*/
 
+    /// @dev Emitted when tokens are transferred from one address to another
     event Transfer(address indexed from, address indexed to, uint256 amount);
 
+    /// @dev Emitted when tokens are transferred from one starknet address to another
+    event TransferStarknet(uint256 indexed from, uint256 indexed to, uint256 amount);
```

Comment:
Those events will need post processing: it can be a starknet address only or a starknet address with a account on kakarot. 
Adding a comment is needed